guava/src/com/google/common/base/Converter.java

```diff
@@ -423,30 +455,30 @@ public boolean equals(@Nullable Object object) {
    * @since 17.0
    */
   public static <A, B> Converter<A, B> from(
-      Function<? super A, ? extends B> forwardFunction,
-      Function<? super B, ? extends A> backwardFunction) {
+      Function<@NonNull ? super @NonNull A, @NonNull ? extends @NonNull B> forwardFunction,
```

Comment:
You don't need to supply both a lower and an upper bound of `@NonNull`; just an upper bound is enough.
(Or did you write them both redundantly on purpose?)